packages/paste-core/components/chat-composer/src/ChatComposer.tsx

```diff
@@ -0,0 +1,156 @@
+  placeholder?: string | JSX.Element;
+  onChange?: OnChangeFunction;
+  maxHeight?: BoxStyleProps['maxHeight'];
+  initialValue?: string;
+
+  /** Mapped to ariaOwneeID on the Lexical ContentEditable component.  */
+  ariaOwns?: ContentEditableProps['ariaOwneeID'];
+  /** Mapped to ariaActiveDescendantID on the Lexical ContentEditable component.  */
+  ariaActiveDescendant?: ContentEditableProps['ariaActiveDescendantID'];
+
+  /** Paste doesn't support using inline styles, use CustomizationProvider instead. */
+  style?: never;
+  /** Paste doesn't support using class names, use CustomizationProvider instead. */
+  className?: never;
+  /** Use ariaOwns instead because it is the name of the actual aria attribute. */
+  ariaOwneeID?: never;
+  /** Use ariaActiveDescendant instead because it is the name of the actual aria attribute. */
+  ariaActiveDescendantID?: never;
+}
+
+export const ChatComposer = React.forwardRef<HTMLDivElement, ChatComposerProps>(
+  (
+    {
+      element = 'CHAT_COMPOSER',
+      onChange,
+      placeholder = '',
+      initialValue,
+      config,
+      maxHeight,
+      ariaOwns,
+      ariaActiveDescendant,
+      ...props
+    },
+    ref
+  ) => {
+    const baseConfigWithEditorState = {
+      ...baseConfig,
+      editorState: initialValue ? () => renderInitialText(initialValue) : undefined,
+    };
+
+    return (
+      <Box
+        flexGrow={1}
+        ref={ref}
+        element={element}
+        position="relative"
+        transition="background-color 100ms ease-in,box-shadow 100ms ease-in,color 100ms ease-in"
+        paddingY="space30"
+        paddingX="space40"
+        borderRadius="borderRadius20"
+        _focusWithin={{boxShadow: 'shadowFocus'}}
+        overflowY="scroll"
+        maxHeight={maxHeight}
+      >
+        <LexicalComposer initialConfig={merge(baseConfigWithEditorState, config)}>
+          <RichTextPlugin
+            contentEditable={
+              <ContentEditable
+                {...props}
+                ariaActiveDescendantID={ariaActiveDescendant}
+                ariaOwneeID={ariaOwns}
+                className="editor-content-editable"
+              />
+            }
+            placeholder={<PlaceholderWrapper element={`${element}_PLACEHOLDER_WRAPPER`} placeholder={placeholder} />}
+            ErrorBoundary={ErrorBoundary}
+          />
+          {onChange ? <OnChangePlugin onChange={onChange} /> : <></>}
+          {/* HistoryPlugin allows for undo/redo */}
+          <HistoryPlugin />
+          {/* AutoLinkPlugin turns text into a anchor when it is a URL or email address */}
+          <AutoLinkPlugin />
+        </LexicalComposer>
+      </Box>
+    );
+  }
+);
+
+ChatComposer.displayName = 'ChatComposer';
+
+ChatComposer.propTypes = {
+  children: PropTypes.element,
+  onChange: PropTypes.func,
+  element: PropTypes.string,
+  config: PropTypes.shape({
+    onError: PropTypes.func.isRequired,
+    namespace: PropTypes.string.isRequired,
+    /*
```

Comment:
OPEN QUESTION: maybe I'm rusty on propTypes, but I'm getting type errors with the editable boolean prop. I'm also not sure how to make the theme and editorState propType correctly.

Reply:
Sorry! The comment here didn't give much context. I had the propType: 
`editable: PropTypes.bool` for a prop where the type is `editable?: boolean`

The error was that the types didn't match. It said the propType was 'boolean | null | undefined' but the prop was 'boolean | undefined'. If I made the propType required, the types stopped complaining but then storybook would complain because I was missing a required propType.

For now, I just removed it because it is not a huge deal.

I did figure out the right way to write prop types for theme and editorState.